--- a/utilities/ovs-ctl.in
+++ b/utilities/ovs-ctl.in
@@ -230,7 +230,11 @@
         fi
 
         # Start ovs-vswitchd.
-        set ovs-vswitchd unix:"$DB_SOCK"
+        set ovs-vswitchd
+        if test X"$DPDK_OPTS" != X; then
+	          set "$@" $DPDK_OPTS --
+	      fi
+	      set "$@" unix:"$DB_SOCK"
         set "$@" -vconsole:emer -vsyslog:err -vfile:info
         if test X"$MLOCKALL" != Xno; then
             set "$@" --mlockall
